types/index.d.ts

```diff
@@ -424,6 +424,7 @@ declare namespace Knex {
   //
   // QueryInterface
   //
+  type Statements = "with" | "select" | "columns" | "where" | "union" | "join" | "group" | "order" | "having" | "limit" | "offset";
```

Comment:
Naming this ClearStatements would be less probably to conflict with some other random type declarations.

Reply:
Other that that this looks what I was expecting 👍